### PR TITLE
Fix: change LabelResourceRuleFormat from Annotations to Labels

### PR DESCRIPTION
**Pull request by chengleqi**
### Description of your changes

Fixes #

I have:

- [x] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [x] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/velaql/providers/query/tree.go [hunk 1]
```diff
@@ -880,8 +880,8 @@ func mergeCustomRules(ctx context.Context, k8sClient client.Client) error {
 			format      string
 			err         error
 		)
-		if item.Annotations != nil {
-			format = item.Annotations[oam.LabelResourceRuleFormat]
+		if item.Labels != nil {
+			format = item.Labels[oam.LabelResourceRuleFormat]
```
Comment (wonderflow) [inline pkg/velaql/providers/query/tree.go:884]: please add a test for this case.

Comment (chengleqi) [reply pkg/velaql/providers/query/tree.go:884]: OK. will do.

**Comment by chengleqi** [comment]
> BTW, we must keep the consistency, so the annotation should also be checked.

I think it maybe a mistake. See [kruise-rollout plugin](https://github.com/kubevela/catalog/blob/d799c26bfbef247e8a019d715b330164f1255882/addons/kruise-rollout/template.cue#L30)

**Comment by wonderflow** [comment]
> > BTW, we must keep the consistency, so the annotation should also be checked.
> 
> I think it maybe a mistake. See [kruise-rollout plugin](https://github.com/kubevela/catalog/blob/d799c26bfbef247e8a019d715b330164f1255882/addons/kruise-rollout/template.cue#L30)

But why it works? \cc @wangyikewxgm

**Comment by chengleqi** [comment]
> > > BTW, we must keep the consistency, so the annotation should also be checked.
> > 
> > 
> > I think it maybe a mistake. See [kruise-rollout plugin](https://github.com/kubevela/catalog/blob/d799c26bfbef247e8a019d715b330164f1255882/addons/kruise-rollout/template.cue#L30)
> 
> But why it works? \cc @wangyikewxgm

It cannot get `oam.LabelResourceRuleFormat` key in Annotations so the [format](https://github.com/kubevela/kubevela/blob/fb45a94bb8ec04691f2fdcb79a3a5c46eac832c1/pkg/velaql/providers/query/tree.go#L884) is "" , so it has the same behavior like `oam.ResourceTopologyFormatYAML` in [label](https://github.com/kubevela/kubevela/blob/fb45a94bb8ec04691f2fdcb79a3a5c46eac832c1/pkg/velaql/providers/query/tree.go#L889).

**Review by wonderflow — COMMENTED** [review]
OK, I've checked that all our addons use label. https://github.com/kubevela/catalog/search?q=rules.oam.dev%2Fresource-format

We don't need to keep the compatible. But let's add one test case to cover the json format.

**Comment by chengleqi** [comment]
> OK, I've checked that all our addons use label. https://github.com/kubevela/catalog/search?q=rules.oam.dev%2Fresource-format
> 
> We don't need to keep the compatible. But let's add one test case to cover the json format.

OK. I'll try.

**Pull request merged**